Framework/Core/src/runDataProcessing.cxx

```diff
@@ -1339,7 +1373,20 @@ int runStateMachine(DataProcessorSpecs const& workflow,
   uv_tcp_t serverHandle;
   serverHandle.data = &serverContext;
   uv_tcp_init(loop, &serverHandle);
+
   driverInfo.port = 8080 + (getpid() % 30000);
+
+  if (getenv("DPL_DRIVER_REMOTE_GUI_PORT")) {
+    try {
+      auto envPort = stoi(std::string(getenv("DPL_DRIVER_REMOTE_GUI_PORT")));
```

Comment:
getenv might return 0 and this will then break.

Reply:
If getenv returns 0 the port number will be based on the PID

Reply:
no, it will be undefined behaviour due to std::string(nullptr).